Shipit dashboard: Use taskcluster hooks to update bug analysis

La0 opened the pull request:
Shipit dashboard is now updated in staging through a Taskcluster hook:

1. the bot has a Python entry point *shipit-bot*
2. it reads a Taskcluster secret directly (when credentials are given) or through taskclusterProxy (after reading the patched /etc/hosts)
3. right now it uses the same docker image as the backend, but we could build 2 images: a lightweight one for the backend (no scipy/libmozdata), and the full one for the analysis.

I added nixpkg cacert as a propagated build input for all docker images, with the corresponding env variable (SSL_CERT_FILE) so that python stdlib modules can check ssl certs (needed for urllib...)

La0 commented:
You can view the [current hook](https://tools.taskcluster.net/hooks/#project-shipit/services-staging-shipit_dashboard-bot) i created, and a [successful task](https://tools.taskcluster.net/task-inspector/#WsQ8aTFRQa6V64CyBhASfg/).

garbas commented on src/shipit_dashboard/shipit_bot/cli.py:
```diff
@@ -0,0 +1,33 @@
+from shipit_bot.sync import BotRemote
+
+
+def main():
```
would it be possible to use [click](http://click.pocoo.org) instead of ``argparse``. to use similar tools along projects.

garbas commented on src/shipit_dashboard/setup.py:
```diff
@@ -33,4 +33,9 @@
     license='MPL2',
+    entry_points={
+        'console_scripts': [
+            'shipit-bot= shipit_bot.cli:main',
```
can a command line be called ``shipit-dashboard-bot``. A reason for this is that this bot is very shipit_dashboard specific and names should reflect that.

Also, why is code placed outside  ``src/shipit_dashboard/shipit_dashboard/`` folder? I would expect even the CLI code be inside there ... even if not used. I would just still all the command line specific folder in ``cli.py`` module

La0 replied:
I moved the code in a separate folder because the shipit_dashboard/__init__.py creates a flask application and uses irrevelant settings for the bot.
These are now 2 separate parts of the same project, but in 2 different contexts (web app & taskcluster hook)

La0 commented:
The mercurial merge checks are now also working in taskcluster tasks, using the cache feature.
First [running task](https://tools.taskcluster.net/task-inspector/#Fi_QBgidTS2KrZZt9jCoVg/0).

@garbas Could you review the PR again now ? I updated the commits messages, binary name and removed argparse. Thanks !
